src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -34,14 +40,25 @@ public class ServiceBusJmsConnectionFactory extends JNDIStorable implements Conn
     // JNDI property names
     private static final String CONNECTION_STRING_PROPERTY = "connectionString";
     private static final String CLIENT_ID_PROPERTY = "clientId";
-    
+    private static final String AUDIENCE = "https://servicebus.azure.net/.default";
+    private static final String AUTHORITY = "https://login.microsoftonline.com/%s/";
     private static final int MaxCustomUserAgentLength = 128;
+    
     private final ServiceBusJmsConnectionFactorySettings settings;
     private volatile boolean initialized;
     private JmsConnectionFactory factory;
     private ConnectionStringBuilder builder;
     private String customUserAgent;
     
+    private String tenantId; 
+    private String aadClientId; 
+    private String clientSecret;
+    private String keyName;
+    private String keyValue;
+    private boolean isToken;
+    private boolean userToken;
```

Comment:
>userToken [](http://example.com/codeflow?start=20&length=9)

Is there a need to handle this separately? Is this mainly for our own testing purposes or we need to handle this scenario of not generating a token using ClientId etc but just using a user suppliedToken?

Reply:
if we do supplied token we do need to do it separately but since we are no longer using it we dont need it